packages/react-components/src/components/styles/ReactionOverlay.style.ts

```diff
@@ -0,0 +1,301 @@
+ */
+export function getReactionStyleBucket(activeSpritesCount: number): IReactionStyleBucket {
+  const index = activeSpritesCount < 3 ? 0 : activeSpritesCount < 5 ? getRandomInt(0, 1) : getRandomInt(0, 2);
+  return ReactionStyleBuckets[index];
+}
+
+/* @conditional-compile-remove(reaction) */
+/**
+ * Return a style bucket for burst scenario
+ * @private
+ */
+export function getReactionBurstStyleBucket(): IReactionStyleBucket {
+  const index = getRandomInt(0, ReactionBurstStyleBuckets.length - 1);
+  return ReactionBurstStyleBuckets[index];
+}
+
+/* @conditional-compile-remove(reaction) */
+/**
+ * @private
+ */
+export const moveFrames = memoizeFunction((maxHeight, travelHeight) =>
+  keyframes({
+    '0%': {
+      transform: `translateY(${maxHeight}px)`
+    },
+    '100%': {
+      transform: `translateY(${travelHeight}px)`
+    }
+  })
+);
+
+/* @conditional-compile-remove(reaction) */
+/**
+ * @private
+ */
+export const moveAnimationStyles = (maxHeight: number, travelHeight: number): React.CSSProperties => {
+  return {
+    animationName: moveFrames(maxHeight, travelHeight),
+    animationFillMode: 'forwards',
+    animationDuration: `4.133s`,
+    animationTimingFunction: 'cubic-bezier(0, 0.83, 0.19, 1.09)'
+  };
+};
+
+/* @conditional-compile-remove(reaction) */
+/**
+ * @private
+ */
+export const opacityTransition = memoizeFunction((maxOpacity) =>
+  keyframes({
+    '0%': {
+      opacity: 0
+    },
+    '31.2%': {
+      opacity: maxOpacity
+    },
+    '67.2%': {
+      opacity: maxOpacity
+    },
+    '100%': {
+      opacity: 0,
+      display: 'none',
+      visibility: 'hidden'
+    }
+  })
+);
+
+/* @conditional-compile-remove(reaction) */
+/**
+ * @private
+ */
+export const opacityAnimationStyles = (maxOpacity: number): React.CSSProperties => {
+  return {
+    animationName: opacityTransition(maxOpacity),
+    animationFillMode: 'forwards',
+    animationDuration: `4.133s`
+  };
+};
+
+/* @conditional-compile-remove(reaction) */
+/**
+ * @private
+ */
+export const spriteFrames = memoizeFunction((numOfFrames, displaySizePx) =>
+  keyframes({
+    from: {
+      backgroundPosition: '0px 0px'
+    },
+    to: {
+      backgroundPosition: `0px -${numOfFrames * displaySizePx}px`
+    }
+  })
+);
+
+/* @conditional-compile-remove(reaction) */
+/**
+ * @private
+ */
+export const spriteAnimationStyles = (
+  numOfFrames: number,
+  displaySizePx: number,
+  imageUrl: string
+): React.CSSProperties => {
+  return {
+    height: `${displaySizePx}px`,
+    width: `${displaySizePx}px`,
+    backgroundImage: `url(${imageUrl})`,
+    backgroundRepeat: 'no-repeat',
+    animationName: spriteFrames(numOfFrames, displaySizePx),
+    animationDuration: `${numOfFrames / 24}s`,
+    animationFillMode: `forwards`,
+    animationIterationCount: 'infinite',
+    animationTimingFunction: `steps(${numOfFrames})`,
+    backgroundSize: `${displaySizePx}px ${numOfFrames * displaySizePx}px`,
+    transform: `scale(${displaySizePx}/128)`
```

Comment:
same here try to use rem instead of px

Reply:
One thing to note on this, normally we can blindly replace px with rem, but as this feature is working with sprite sheets we have to be really careful what is in rem and what is in px.
Sprite sheets contain a fixed number of images of fixed pixel size, so when animating sprites we have to work in pixels (but just for the sprite animation). We should be able to use rem for any other size values but we need to be super careful that they don't break the animation

Reply:
I agree. Also the transform functions don't always mean that we are rendering in pixels. It is the pixel height of the sprite image. We have to use px in those regards as animation frames will have to move in pixels rather than rem.